### DISCUSSION TITLE
Styling - CalendarDatePicker - Focus change BorderElement

### DISCUSSION THREAD

**Discussion by marcosgerene** [AvaloniaUI/Avalonia#13154] [Q&A] [answered]
Hi folks,

Some days ago I asked for help to change BorderThickness of a Textbox when it is focused. Q&A: https://github.com/AvaloniaUI/Avalonia/discussions/13054

So I have found a solution that works fine:

```
<Style Selector="TextBox.Padrao">
	<Setter Property="Padding" Value="5 6 0 0"/>
	<Setter Property="BorderThickness" Value="0 0 0 1"/>
	<Setter Property="CornerRadius" Value="0 0 0 0"/>		
	<Setter Property="MinWidth" Value="60" />
	<Setter Property="FontWeight" Value="Normal" />
	<Setter Property="FontSize" Value="15" />
	<Setter Property="Background" Value="{DynamicResource ControlBackground}" />
</Style>

<Style Selector="TextBox.Padrao:focus /template/ Border#PART_BorderElement">
	<Setter Property="BorderBrush" Value="#455a64"/>
	<Setter Property="BorderThickness" Value="0 0 0 3"/>
</Style>
```
I want to do the same with ```CalendarDatePicker``` but it is not working.

```
<Style Selector="CalendarDatePicker.Padrao">
	<Setter Property="Padding" Value="5 6 0 0"/>
	<Setter Property="BorderThickness" Value="0 0 0 1"/>
	<Setter Property="CornerRadius" Value="0 0 0 0"/>
	<Setter Property="MinWidth" Value="60" />
	<Setter Property="FontWeight" Value="Normal" />
	<Setter Property="FontSize" Value="15" />
	<Setter Property="Background" Value="{DynamicResource ControlBackground}" />
</Style>

<Style Selector="CalendarDatePicker.Padrao:focus /template/ Border#PART_BorderElement">
	<Setter Property="BorderBrush" Value="#455a64"/>
	<Setter Property="BorderThickness" Value="0 0 0 3"/>
</Style>
```

Look to the result (Thickness on focus):
![focus](https://github.com/AvaloniaUI/Avalonia/assets/15945003/594320fa-84f0-4643-8f2b-04cefa8e2319)

Could someone show me how to fix this?

**Comment by timunie** [comment, marked as answer]
In source code, it's always worth to check orginal styles to understand which selector should be changed. 

https://github.com/AvaloniaUI/Avalonia/blob/4ef901b22b9c8102bdfdcb34357fc34209af0b57/src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml#L167-L171

you can also override the used DynamicResouce in App.Resources: 
```xml
<Thickness x:Key="TextControlBorderThemeThicknessFocused">0 0 0 3</Thickness>
```

Btw: I like your design ❤️

**Comment by marcosgerene** [comment]
@timunie 


> In source code, it's always worth to check orginal styles to understand which selector should be changed.

In this case what should be the selector? 
I tried ```CalendarDatePicker.Padrao:focus /template/ Border#Background``` and did not work, where am I wrong?


> ``` <Thickness x:Key="TextControlBorderThemeThicknessFocused">0 0 0 3</Thickness> ```

This second aproach worked fine, thanks!  (... but I still want to understand the first one).


> Btw: I like your design ❤️

Thank you =D

**Reply by timunie** [reply]
guess selector should be: `CalendarDatePicker.Padrao:focus-within:not(CalendarDatePicker:focus) /template/ Border#Background`

as a rule of thumb, I simply replace ^ by the control type and class name, that should work.

**Comment by marcosgerene** [comment]
> as a rule of thumb, I simply replace ^ by the control type and class name, that should work.

It helped a lot!
